bluemira/equilibria/coils/_grouping.py

```diff
@@ -1184,10 +1184,143 @@ def set_optimisation_state(
         """
         cc = self.get_control_coils()
         if opt_currents is not None:
-            cc.current = opt_currents * current_scale
+            cc._opt_currents = opt_currents * current_scale
         if coil_position_map is not None:
             cc._set_opt_positions(coil_position_map)
 
+    @property
+    def n_current_optimisable_coils(self) -> int:
+        """
+        Get the number of all current optimisable coils
+        """
+        return len(self.current_optimisable_coil_names)
+
+    @property
+    def current_optimisable_coil_names(self) -> list[str]:
+        """
+        Get the names of all current optimisable coils
+        """
+        optimisable_coil_names = [
+            c.primary_coil.name if isinstance(c, Circuit) else c.name
+            for c in self._coils
+        ]
+        return [*flatten_iterable(optimisable_coil_names)]
+
+    @property
+    def all_current_optimisable_coils(self) -> list[Coil]:
+        """
+        Get the names of all coils that can be current optimised.
+        """
+        return [self[cn] for cn in self.current_optimisable_coil_names]
+
+    def get_current_optimisable_coils(
+        self, coil_names: list[str] | None = None
+    ) -> list[Coil]:
+        """
+        Get the coils that can be current optimised.
```

Comment:
Should this be changed to something like 'get list of selected coils from coils that can be current optimised'?